litestar/connection/base.py

```diff
@@ -115,7 +126,8 @@ def url(self) -> URL:
             A URL instance constructed from the request's scope.
         """
         if self._url is Empty:
-            self._url = self.scope["_url"] = URL.from_scope(self.scope)  # type: ignore[typeddict-unknown-key]
+            self._url = URL.from_scope(self.scope)
```

Comment:
Why don't we adapt the same thing we have in the `Headers.from_scope` method? It feels a bit nicer to me that all those methods kinda follow the same logic

Reply:
Sorry, can you demonstrate what you mean?

Reply:
The `Headers.from_scope` classmethod contains the "cache in scope" functionality:

https://github.com/litestar-org/litestar/blob/a2e5b7854edce5b40d23bc3493d47f579bdcddef/litestar/datastructures/headers.py#L74-L88

Since we already have `URL.from_scope`, I think it would make sense that it works in the same way. Or we make `Headers.from_scope` not store itself in the scope; Either way, IMO it's a good idea if they both do the same thing :)

Reply:
OK I follow and agree, we'll use the `from_scope()` methods to also do the caching of the values in scope where required :+1:

Reply:
Thinking about this more, we'd have to add a `scope_cache_key` parameter to the `from_scope()` methods, b/c we cache URLs under both `url` and `base_url` keys.

Reply:
> Thinking about this more, we'd have to add a `scope_cache_key` parameter to the `from_scope()` methods, b/c we cache URLs under both `url` and `base_url` keys.

Those would be the same though if constructed from the same scope, right?

Reply:
In `base_url()` we unpack the connection scope into a new scope, modify the path, query etc and construct the url with `from_scope()`. Even though they are separate scopes, they share the same state, and now that we are caching in state, the first one to access would be the one that is cached and returned for either `url()` or `base_url()`. 

I'm AFK though, so I've not actually ran the scenario but I think that's right, isn't it?

Reply:
> I'm AFK though, so I've not actually ran the scenario but I think that's right, isn't it?

Same here but that sounds plausible!

Reply:
I'm going to leave this as is for now, so that this PR is only changing where we cache stuff within scope, which is its original intent.

Changing the behavior of either `URL.from_scope()` or `Headers.from_scope()` should be looked at, but probably best in its own PR as it will likely need some discussion around it.